pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunctionUtils.java

```diff
@@ -181,37 +162,63 @@ public static int parsePercentile(String percentileString) {
+   * Collects all transform expressions required for aggregation/group-by queries.
+   * <p>NOTE: We don't need to consider order-by columns here as the ordering is only allowed for aggregation functions
+   *          or group-by expressions.
    */
-  public static Set<TransformExpressionTree> collectExpressionsToTransform(BrokerRequest brokerRequest,
-      AggregationFunctionContext[] functionContexts) {
-
-    Set<TransformExpressionTree> expressionTrees = new LinkedHashSet<>();
-    for (AggregationFunctionContext functionContext : functionContexts) {
-      AggregationFunction function = functionContext.getAggregationFunction();
-      expressionTrees.addAll(function.getInputExpressions());
+  public static Set<TransformExpressionTree> collectExpressionsToTransform(AggregationFunction[] aggregationFunctions,
+      @Nullable TransformExpressionTree[] groupByExpressions) {
+    Set<TransformExpressionTree> expressions = new HashSet<>();
+    for (AggregationFunction aggregationFunction : aggregationFunctions) {
+      //noinspection unchecked
+      expressions.addAll(aggregationFunction.getInputExpressions());
+    }
+    if (groupByExpressions != null) {
+      expressions.addAll(Arrays.asList(groupByExpressions));
     }
+    return expressions;
+  }
 
-    // Extract group-by expressions
-    if (brokerRequest.isSetGroupBy()) {
-      for (String expression : brokerRequest.getGroupBy().getExpressions()) {
-        expressionTrees.add(TransformExpressionTree.compileToExpressionTree(expression));
-      }
+  /**
+   * Creates a map from expression required by the {@link AggregationFunction} to {@link BlockValSet} fetched from the
+   * {@link TransformBlock}.
+   */
+  public static Map<TransformExpressionTree, BlockValSet> getBlockValSetMap(AggregationFunction aggregationFunction,
+      TransformBlock transformBlock) {
+    //noinspection unchecked
+    List<TransformExpressionTree> expressions = aggregationFunction.getInputExpressions();
+    int numExpressions = expressions.size();
+    if (numExpressions == 0) {
+      return Collections.emptyMap();
     }
+    if (numExpressions == 1) {
+      TransformExpressionTree expression = expressions.get(0);
+      return Collections.singletonMap(expression, transformBlock.getBlockValueSet(expression));
+    }
+    Map<TransformExpressionTree, BlockValSet> blockValSetMap = new HashMap<>();
+    for (TransformExpressionTree expression : expressions) {
+      blockValSetMap.put(expression, transformBlock.getBlockValueSet(expression));
+    }
+    return blockValSetMap;
+  }
 
-    // TODO: Add order-by expressions when available in brokerRequest for aggregation queries.
-    // The current order-by implementation assumes that ordering will be on aggregation/group-by columns.
-    return expressionTrees;
+  /**
+   * Creates a map from expression required by the {@link AggregationFunctionColumnPair} to {@link BlockValSet} fetched
+   * from the {@link TransformBlock} (for star-tree).
```

Comment:
(nit) - For better readability, consider putting "for star tree" at the beginning something like 'This function is used in start tree code path only`

Reply:
Done